tests/platform/test_link_flap.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+import pytest
+
+from common.platform.device_utils import fanout_switch_port_lookup
+from common.utilities import wait_until
+
+class TestLinkFlap:
+    def __get_dut_if_status(self, dut, ifname=None):
+        if not ifname:
+            status = dut.show_interface(command='status')['ansible_facts']['int_status']
+        else:
+            status = dut.show_interface(command='status', interfaces=[ifname])['ansible_facts']['int_status']
+
+        return status
+
+
+    def __check_if_status(self, dut, dut_port, exp_state):
+        status = self.__get_dut_if_status(dut, dut_port)[dut_port]
+        return status['oper_state'] == exp_state
+
+
+    def __toggle_one_link(self, dut, dut_port, fanout, fanout_port):
+        logging.info("Testing link flap on {}".format(dut_port))
+
+        status = self.__get_dut_if_status(dut, dut_port)[dut_port]
+        logging.debug("Dut port status {}".format(status))
+        assert status['oper_state'] == 'up', "Skipping dut port {}: link operational down".format(status)
+
+        logging.info("Shutting down fanout switch {} port {} connecting to {}".format(fanout.hostname, fanout_port, dut_port))
+        self.ports_shutdown_by_test.add((fanout, fanout_port))
+        fanout.shutdown(fanout_port)
+        wait_until(30, 1, self.__check_if_status, dut, dut_port, 'down')
+        status = self.__get_dut_if_status(dut, dut_port)[dut_port]
+        logging.debug("Interface fact  : {}".format(status))
+        assert status['oper_state'] == 'down', "dut port {} didn't go down as expected".format(dut_port)
+
+        logging.info("Bring up fanout switch {} port {} connecting to {}".format(fanout.hostname, fanout_port, dut_port))
+        fanout.no_shutdown(fanout_port)
+        wait_until(30, 1, self.__check_if_status, dut, dut_port, 'up')
+        status = self.__get_dut_if_status(dut, dut_port)[dut_port]
+        logging.debug("Interface fact  : {}".format(status))
+        assert status['oper_state'] == 'up', "dut port {} didn't come up as expected".format(dut_port)
+        self.ports_shutdown_by_test.discard((fanout, fanout_port))
+
+
+    def __build_test_candidates(self, dut, fanouthosts):
+        status = self.__get_dut_if_status(dut)
+        candidates = []
```

Comment:
If candidates is somehow empty at the end should we `skip` and/or print a message to the user? If I'm following this correctly it looks like this will just silently pass if no candidates are found.

Reply:
Good idea. Add skip message.